src/pydvl/utils/numeric.py

```diff
@@ -113,6 +124,57 @@ def random_powerset(
         total += 1
 
 
+def random_powerset_group_conditional(
+    s: NDArray[T],
+    labels: NDArray[np.int_],
+    n_samples: Optional[int] = None,
+    min_elements: int = 1,
+) -> Generator[NDArray[T], None, None]:
+    """
+    :param s: set to sample from
+    :param labels: Labels that a minimum number from each label group is included.
+    :param n_samples: if set, stop the generator after this many steps.
+        Defaults to `np.iinfo(np.int32).max`
+    :param min_elements: The minimum number of elements for each distinct label and its associated group.
+
+    :return: Samples from the power set of s
+    :raises: TypeError: if the data `s` is not a NumPy array
+    :raises: ValueError: if the element sampling probability is not in [0,1]
+
+    """
+    if not isinstance(s, np.ndarray):
+        raise TypeError("Set must be an NDArray")
+
+    if not isinstance(labels, np.ndarray):
+        raise TypeError("Labels must be an NDArray")
+
+    if len(labels) != len(s):
+        raise ValueError("Set and labels have to be of same size.")
+
+    rng = np.random.default_rng()
+    total = 1
+    if n_samples is None:
+        n_samples = np.iinfo(np.int32).max
+
+    unique_labels = np.unique(labels)
+    while total <= n_samples:
+
```

Comment:
I think we need to remove `n_samples` from all samplers and have infinite generators. After removing the parameter, documentation, etc.:

```suggestion
    while True:
```
(also, no blank line after `while`)